### ISSUE TITLE
TransactionOnNetwork doesn't contain smart contract results in the npm package

### ISSUE THREAD

**Issue by MWFIAE** [ElrondNetwork/elrond-sdk-erdjs#96]
Hi everyone,

sadly this is still an issue:

While the code in the repo looks (and probably works) fine. 
https://github.com/ElrondNetwork/elrond-sdk-erdjs/blob/main/src/transactionOnNetwork.ts#L73
```typescript
transactionOnNetwork.smartContractResults = SmartContractResults.fromHttpResponse(response.smartContractResults || []);
```

The version published on npm contains an error:
```typescript
transactionOnNetwork.results = smartContractResults_1.SmartContractResults.fromHttpResponse(response.results || []);
```

As you can see it tries to use the results field instead of the smartContractResults field. Resulting in no results being included in the transaction.

**Comment by raress96** [comment]
This might actually be intended. I use the Elrond API instead of the Elrond Gateway, so I get the transaction using the ApiProvider getTransaction method, and it works fine since the API has the field as `results`, but the gateway has the field like `smartContractResults`. Maybe this should be compatible with both approaches?

**Comment by MWFIAE** [comment]
If this is intended the `getAsOnNetwork` method of the `Transaction` class should take an `IApiProvider` instead of an `IProvider`. Since the ApiProvider only implements `IApiProvider` and not the `IProvider` interface.

Also it might be a good idea to actually document that somewhere :)

So I will keep this issue open until this is changed or compatible with both providers :) (And in the meantime I will try to use the ApiProvider with some crazy casting so I can use CI/CD again 😄 )

**Comment by MWFIAE** [comment]
@andreibancioiu Any news on this?
It's a huge issue imho :)

**Comment by MWFIAE** [comment]
@ccorcoveanu 
I have seen that you recently merged a pr that contained the changes for the code in question.

Can you please address the issues as mentioned above? 🙏

**Comment by andreibancioiu** [comment]
@MWFIAE, thank you for bringing this up and for the detailed analysis :pray: 

We are working on a fix.

**Comment by andreibancioiu** [comment]
@MWFIAE, we have included a partial fix in version `9.0.3` - please see description of https://github.com/ElrondNetwork/elrond-sdk-erdjs/pull/123.

This issue can remain open, until we come with a complete fix (and unify `ProxyProvider` and `ApiProvider`).

For the moment (not for long), if using `transaction.getAsOnNetwork()`, passing a `ProxyProvider` is preferred if you are interested into smart contract results as well.

**Comment by andreibancioiu** [comment]
The issue should be resolved in erdjs 10 :pray: 

Please see:
 - https://docs.elrond.com/sdk-and-tools/erdjs/erdjs-cookbook/#parsing-contract-results
 - https://docs.elrond.com/sdk-and-tools/erdjs/erdjs-migration-guides/
 - https://docs.elrond.com/sdk-and-tools/erdjs/extending-erdjs/#extending-a-default-network-provider
